components/PLP/CategoryResult.tsx

```diff
@@ -0,0 +1,169 @@
+import { SendEventOnLoad } from "$store/components/Analytics.tsx";
+import { Layout as CardLayout } from "$store/components/product/ProductCard.tsx";
+import Icon from "$store/components/ui/Icon.tsx";
+import GalleryControls from "$store/islands/GalleryControls.tsx";
+import { Result } from "$store/components/search/SearchResult.tsx"
+import { useOffer } from "$store/sdk/useOffer.ts";
+import type { ProductListingPage } from "apps/commerce/types.ts";
+import { mapProductToAnalyticsItem } from "apps/commerce/utils/productToAnalyticsItem.ts";
+import ProductGallery from "../product/ProductGallery.tsx";
+import NotFound from "$store/components/search/NotFound.tsx";
+import type { PropsNotFound } from "$store/components/search/NotFound.tsx"
+import type { SectionProps } from "deco/types.ts";
+import type { Section } from "$store/components/search/PhotoAndProducts.tsx"
+
+export interface Layout {
+  /**
+   * @description Use drawer for mobile like behavior on desktop. Aside for rendering the filters alongside the products
+   */
+  variant?: "aside" | "drawer";
+  /**
+   * @description Number of products per line on grid
+   */
+}
+
+/**
+ * @titleBy category
+ */
+export interface CategoryArray{
+  /** @description RegExp to enable this banner on the current URL. Use /feminino/* to display this banner on feminino category  */
+  matcher: string;
+  category: string;
+  url: string
+  /**
+ * @description Subcategorias existentes
+ */
+  items?: { 
+    label: string; 
+    url: string
+  }[]; 
+}
+
+export interface Props {
+  /** @title Integration */
+  categories: CategoryArray[];
+  page: ProductListingPage | null;
```

Comment:
O que diz para uma tipagem que é um array é o `[]` entao nomear CategoryArray[] é ambiguo.
```suggestion
export interface Category {
  /** @description RegExp to enable this banner on the current URL. Use /feminino/* to display this banner on feminino category  */
  matcher: string;
  category: string;
  url: string
  /**
 * @description Subcategorias existentes
 */
  items?: { 
    label: string; 
    url: string
  }[]; 
}

export interface Props {
  /** @title Integration */
  categories: Category[];
  page: ProductListingPage | null;
```